libfwupdplugin/fu-ble-device.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2021 Ricardo Cañuelo <ricardo.canuelo@collabora.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN				"FuBleDevice"
+
+#include "config.h"
+
+#include "fu-device-private.h"
+#include "fu-ble-device.h"
+
+
+/**
+ * SECTION:fu-ble-device
+ * @short_description: a Bluetooth LE device
+ *
+ * An object that represents a Bluetooth LE device.
+ *
+ * See also: #FuDevice
+ */
+
+typedef struct {
+	gchar		*name;
+	gchar		*address;
+	gchar		*adapter;
+	GHashTable	*characteristics;	/* utf8 : utf8 */
+} FuBleDevicePrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (FuBleDevice, fu_ble_device, FU_TYPE_DEVICE)
+
+#define GET_PRIVATE(o) (fu_ble_device_get_instance_private (o))
+
+/**
+ * fu_ble_device_get_name:
+ * @self: A #FuBleDevice
+ *
+ * Gets the name of the device.
+ *
+ * Returns: The name of the device.
+ *
+ * Since: 1.5.7
+ **/
+const gchar *
+fu_ble_device_get_name (FuBleDevice *self)
+{
+	FuBleDevicePrivate *priv = GET_PRIVATE (self);
+
+	return priv->name;
+}
+
+/**
+ * fu_ble_device_get_address:
+ * @self: A #FuBleDevice
+ *
+ * Gets the address of the device.
```

Comment:
I think an example expected syntax would be good to leave here in documentation

Reply:
Done.